### PR TITLE
Avoid warnings in pyarrow and msgpack

### PR DESCRIPTION
**Pull request by mrocklin**

### CODE REVIEW

File: distributed/protocol/arrow.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ def serialize_batch(batch):
     writer = pyarrow.RecordBatchStreamWriter(sink, batch.schema)
     writer.write_batch(batch)
     writer.close()
-    buf = sink.get_result()
+    buf = sink.getvalue()
```
Comment (xhochy) [inline distributed/protocol/arrow.py]: You might need to increase the minimal Arrow version for this.

Comment (mrocklin) [reply distributed/protocol/arrow.py]: Which version would you recommend?  Also, is that version commonly deployed?  My guess is that current Arrow users are early adopters and are ok with updating frequently.

Comment (mrocklin) [reply distributed/protocol/arrow.py]: It looks like this works with pyarrow 0.9.0, which seems fairly old to me today

Comment (xhochy) [reply distributed/protocol/arrow.py]: I would suggest to require 0.10 as minimal version which is in `pyarrow` terms already old.

**Comment by mrocklin** [comment]
TLS CI failures are unrelated.  Raised at #2369 

Merging

**Pull request merged**